leshan-client-core/src/main/java/org/eclipse/leshan/client/servers/ServersInfoExtractor.java

```diff
@@ -110,7 +110,34 @@ public static ServersInfo getInfo(Map<Integer, LwM2mObjectEnabler> objectEnabler
                     info.serverUri = new URI((String) security.getResource(SEC_SERVER_URI).getValue());
                     info.serverId = (long) security.getResource(SEC_SERVER_ID).getValue();
                     info.secureMode = getSecurityMode(security);
-                    if (info.secureMode == SecurityMode.PSK) {
+
+                    // find instance id of the associated oscore object (if any)
+                    ObjectLink oscoreObjectLink = (ObjectLink) security.getResource(SEC_OSCORE_SECURITY_MODE).getValue();
+                    int oscoreObjectInstanceId = oscoreObjectLink.getObjectInstanceId();
+                    int oscoreObjectId = oscoreObjectLink.getObjectId();
+                    boolean useOscore = false;
+                    if(oscoreObjectId == OSCORE) {
+                    	useOscore = true;
+                    }
+
+                    if(useOscore) {
+                    	// search corresponding oscore object
+                    	LwM2mObjectInstance oscoreInstance = null;
+                        for (LwM2mObjectInstance oscore : oscores.getInstances().values()) {
```

Comment:
Not need to iterate, there is a getter by Id => `oscores.getInstance(id)`

Reply:
Yes, that is much nicer. Fixed in commit a255bd64d5334790597db3d9de6addcf81b78433